src/gui/qgsnewgeopackagelayerdialog.cpp

```diff
@@ -192,7 +192,8 @@ void QgsNewGeoPackageLayerDialog::mAddAttributeButton_clicked()
   if ( !mFieldNameEdit->text().isEmpty() )
   {
     QString myName = mFieldNameEdit->text();
-    if ( myName == mFeatureIdColumnEdit->text() )
+    QString featureId = mFeatureIdColumnEdit->text();
+    if ( ( featureId.isEmpty() && myName == mFeatureIdColumnEdit->placeholderText() ) || myName == featureId )
```

Comment:
It's a little pedantic, but instead of checking against the placeholderText here could you instead do something like:

    #define DEFAULT_OGR_FID_COLUMN_TITLE "fid" // default value from OGR

...

    mFeatureIdColumnEdit->setPlaceholderText( DEFAULT_OGR_FID_COLUMN_TITLE );

...

    if ( ( featureId.isEmpty() && myName == DEFAULT_OGR_FID_COLUMN_TITLE ) || myName == featureId )

?

I think we probably also want a case-insensitive check here:

    const QString featureId = mFeatureIdColumnEdit.text().isEmpty() ? DEFAULT_OGR_FID_COLUMN_TITLE : mFeatureIdColumnEdit.text();
    if ( myName.compare( featureId, Qt::CaseInsensitive ) == 0 )
    ...

because we don't want users creating fields